internal/catalog/internal/types/failover_policy.go

```diff
@@ -27,7 +33,208 @@ func RegisterFailoverPolicy(r resource.Registry) {
 	r.Register(resource.Registration{
 		Type:     FailoverPolicyV1Alpha1Type,
 		Proto:    &pbcatalog.FailoverPolicy{},
-		Validate: nil,
-		Mutate:   nil,
+		Mutate:   MutateFailoverPolicy,
+		Validate: ValidateFailoverPolicy,
 	})
 }
+
+func MutateFailoverPolicy(res *pbresource.Resource) error {
+	var failover pbcatalog.FailoverPolicy
+
+	if err := res.Data.UnmarshalTo(&failover); err != nil {
+		return resource.NewErrDataParse(&failover, err)
+	}
+
+	// Handle eliding empty configs.
+	if failover.Config != nil && failover.Config.IsEmpty() {
+		failover.Config = nil
+	}
+	for port, pc := range failover.PortConfigs {
+		if pc.IsEmpty() {
+			delete(failover.PortConfigs, port)
+		}
+	}
+	if len(failover.PortConfigs) == 0 {
+		failover.PortConfigs = nil
+	}
+
+	// TODO(rb): normalize dest ref tenancies
+
+	return res.Data.MarshalFrom(&failover)
+}
+
+func ValidateFailoverPolicy(res *pbresource.Resource) error {
+	var failover pbcatalog.FailoverPolicy
+
+	if err := res.Data.UnmarshalTo(&failover); err != nil {
+		return resource.NewErrDataParse(&failover, err)
+	}
+
+	var merr error
+
+	if res.Id != nil && res.Id.Tenancy != nil {
+		if res.Id.Tenancy.PeerName != "local" {
+			merr = multierror.Append(merr, resource.ErrInvalidField{
+				Name: "id",
+				Wrapped: resource.ErrInvalidField{
+					Name: "tenancy",
+					Wrapped: resource.ErrInvalidField{
+						Name:    "peer_name",
+						Wrapped: fmt.Errorf("must be local"),
+					},
+				},
+			})
+		}
+	}
```

Comment:
Eventually this will be enforced by the resource service. For the sake of not having more to remember to rip out in the future, maybe this should be removed?